test/lib/Elastica/Test/Query/HasParentTest.php

```diff
@@ -52,4 +56,53 @@ public function testSetScope()
 
         $this->assertEquals($expectedArray, $query->toArray());
     }
+
+    /**
+     * @group functional
+     */
+    public function testHasParent()
+    {
+        $index = $this->_createIndex();
+
+        $shopType = $index->getType('shop');
+        $productType = $index->getType('product');
+        $mapping = new Mapping();
+        $mapping->setParent('shop');
+        $productType->setMapping($mapping);
+
+
+        $shopType->addDocument(new Document('zurich', array('brand' => 'google')));
+        $shopType->addDocument(new Document('london', array('brand' => 'apple')));
+
+
+        $doc1 = new Document(1, array('device' => 'chromebook'));
+        $doc1->setParent('zurich');
+
+        $doc2 = new Document(2, array('device' => 'macmini'));
+        $doc2->setParent('london');
+
+
+        $productType->addDocument($doc1);
+        $productType->addDocument($doc2);
```

Comment:
I actually used here addDocument on purpose in case some parent mapping goes wrong or is incorrect, it is easier to find where the exception was thrown.